core/Symbols.cc

```diff
@@ -812,6 +812,23 @@ SymbolRef Symbol::singletonClass(GlobalState &gs) {
     singletonInfo->setSuperClass(Symbols::todo());
     singletonInfo->setIsModule(false);
 
+    auto tp = gs.enterTypeMember(this->loc(), singleton, Names::Constants::AttachedClass(), Variance::CoVariant);
+
+    // Initialize the bounds of AttachedClass as top and bottom, as the upper
+    // bound will be updated to the result of `externalType` in the resolver's
+    // global pass.
```

Comment:
I don't think that it happens in global pass from what I see in this PR—looks like it's in ResolveSignaturesWalk?

Reply:
Originally I was fixing it in the GlobalPass, but I must have forgotten to update the comment after moving that functionality. Thanks for catching this!